Tools/i18n/pygettext.py

```diff
@@ -212,22 +213,39 @@ def escape_ascii(s, encoding):
 def escape_nonascii(s, encoding):
     return ''.join(escapes[b] for b in s.encode(encoding))
 
+# Split a string according to whitespaces and keep
+# the whitespaces in the resulting array thanks to
+# the capturing group.
+_space_splitter = re.compile(r'(\s+)').split
 
-def normalize(s, encoding):
+def normalize(s, encoding, options):
     # This converts the various Python string types into a format that is
-    # appropriate for .po files, namely much closer to C style.
-    lines = s.split('\n')
-    if len(lines) == 1:
-        s = '"' + escape(s, encoding) + '"'
-    else:
-        if not lines[-1]:
-            del lines[-1]
-            lines[-1] = lines[-1] + '\n'
-        for i in range(len(lines)):
-            lines[i] = escape(lines[i], encoding)
-        lineterm = '\\n"\n"'
-        s = '""\n"' + lineterm.join(lines) + '"'
-    return s
+    # appropriate for .po files, namely much closer to C style,
+    # while wrapping to options.width.
+    lines = []
+    for line in s.splitlines(True):
+        if len(escape(line, encoding)) > options.width and ' ' in line: # don't wrap single words
```

Comment:
Take into account quotes for not wrapped line.

Take into account the length of the prefix (like `'msgid_plural'`) for single not wrapped line.